shared/src/TwinCollection.cs

```diff
@@ -176,11 +177,11 @@ public Metadata GetMetadata()
         /// Gets the LastUpdated time for this property
         /// </summary>
         /// <returns>DateTime instance representing the LastUpdated time for this property</returns>
-        /// <exception cref="System.ArgumentNullException">Thrown when the TwinCollection metadata is null. 
+        /// <exception cref="System.NullReferenceException">Thrown when the TwinCollection metadata is null. 
```

Comment:
Reverting this change so no more behavior changes. It will still throw nre as before, we are documenting it.